utilities/test_suite/HIP/Tensor_hip.cpp

```diff
@@ -610,6 +614,147 @@ int main(int argc, char **argv)
+                    memcpy(permutedArray, initialPermuteArray, batchSize * sizeof(Rpp32u));
+                    randomize(initialPermuteArray, batchSize);
+                    memcpy(permutedArray + batchSize, initialPermuteArray, batchSize * sizeof(Rpp32u));
+                    randomize(initialPermuteArray, batchSize);
+                    memcpy(permutedArray + (batchSize * 2), initialPermuteArray, batchSize * sizeof(Rpp32u));
+                    randomize(initialPermuteArray, batchSize);
+                    memcpy(permutedArray + (batchSize * 3), initialPermuteArray, batchSize * sizeof(Rpp32u));
+                }
+
+                for (uint i = 0, j = 0; j < batchSize * 4; i++, j += 4)
+                {
+                    permutationTensor[j] = permutedArray[i];
+                    permutationTensor[j + 1] = permutedArray[i + batchSize];
+                    permutationTensor[j + 2] = permutedArray[i + (batchSize * 2)];
+                    permutationTensor[j + 3] = permutedArray[i + (batchSize * 3)];
+                }
+
+                RpptROI *roiPtrInputCropRegion;
+                hipHostMalloc(&roiPtrInputCropRegion, 4 * sizeof(RpptROI));
+
+                if(qaFlag)
+                {
+                    roiPtrInputCropRegion[0].xywhROI.roiWidth = (int)std::round(((int)std::round(0.3 * iX) >> 3)) << 3;                 // w1
+                    roiPtrInputCropRegion[0].xywhROI.roiHeight = (int)std::round(((int)std::round(0.5 * iY) >> 3)) << 3;                // h1
+                    roiPtrInputCropRegion[1].xywhROI.roiWidth = iX - roiPtrInputCropRegion[0].xywhROI.roiWidth;   // w2
+                    roiPtrInputCropRegion[1].xywhROI.roiHeight = roiPtrInputCropRegion[0].xywhROI.roiHeight;      // h2
+                    roiPtrInputCropRegion[2].xywhROI.roiWidth = roiPtrInputCropRegion[0].xywhROI.roiWidth;        // w3
+                    roiPtrInputCropRegion[2].xywhROI.roiHeight = iY - roiPtrInputCropRegion[0].xywhROI.roiHeight; // h3
+                    roiPtrInputCropRegion[3].xywhROI.roiWidth = iX - roiPtrInputCropRegion[0].xywhROI.roiWidth;   // w4
+                    roiPtrInputCropRegion[3].xywhROI.roiHeight = iY - roiPtrInputCropRegion[0].xywhROI.roiHeight; // h4
+                    roiPtrInputCropRegion[0].xywhROI.xy.x = iX - roiPtrInputCropRegion[0].xywhROI.roiWidth;  // x1
+                    roiPtrInputCropRegion[0].xywhROI.xy.y = 0; // y1
+                    roiPtrInputCropRegion[1].xywhROI.xy.x = iX - roiPtrInputCropRegion[1].xywhROI.roiWidth;  // x2
+                    roiPtrInputCropRegion[1].xywhROI.xy.y = 0; // y2
+                    roiPtrInputCropRegion[2].xywhROI.xy.x = 0;  // x3
+                    roiPtrInputCropRegion[2].xywhROI.xy.y = iY - roiPtrInputCropRegion[2].xywhROI.roiHeight; // y3
+                    roiPtrInputCropRegion[3].xywhROI.xy.x = 0;  // x4
+                    roiPtrInputCropRegion[3].xywhROI.xy.y = iY - roiPtrInputCropRegion[3].xywhROI.roiHeight; // y4
```

Comment:
Please precommpute wherever possible and change these initializations to be more concise
https://github.com/r-abishek/rpp/blob/250d99db909d0d645e7f3487fbd90b555d85b8f2/utilities/examples/brightness/brightness_hip.cpp#L167

Reply:
sure